packages/dhis2/src/Adaptor.js

```diff
@@ -236,13 +236,30 @@ export function create(resourceType, data, options = {}, callback = false) {
     const { params, requestConfig } = resolvedOptions;
     const { configuration } = state;
 
-    return request(configuration, {
-      method: 'post',
-      url: generateUrl(configuration, resolvedOptions, resolvedResourceType),
-      params,
-      data: nestArray(resolvedData, resolvedResourceType),
-      ...requestConfig,
-    }).then(result => {
+    let promise;
+    if (shouldUseNewTracker(resolvedResourceType)) {
```

Comment:
hmm, hang on, are we sure `request` should use the new tracker?

Request is a low level API and it should do _exactly_ what you tell it. If you do `request('trackedEntityInstance')` it should return an error.

Oh, also request is a private function, not used directly in job code. So we really shouldn't be adding intelligence to it.

Reply:
@josephjclark  do you mind expounding on this?

Reply:
`request()` is a basic HTTP request API. It should call whatever URL you pass it. It should not try to be backwards compatible, and should not try to map URLs for you.

The new tracker stuff should ONLY apply to higher-order functions like `create` and `update`, which abstract the URL from the user.

Please remove `if(shouldUseNewTracker(resolvedResourceType)` from this function!

Reply:
@josephjclark  I only used the new tracker on `upsert`, `create` , and `update` not any other function. The code you are pointing to above is for the `create` function

Reply:
Ah you are right! I'm sorry, I was confused by the diff